### PR TITLE
HTML viz updates

### PR DESCRIPTION
**Pull request by JenkeScheen**
## Description
Some changes to the HTML viz:
- [x] add NS2B3 correct metadata
- [x] add NS2B3 correct handling of chain A and B
- [x] add NS2B3 correct depiction of all ligands in case of multi-SDF input
- [x] add more key info as drop-down in HTML view
- [x] add information about mutants on hover in HTML view

## Developers certificate of origin
- [x] I certify that this contribution is covered by the MIT license as defined in our [LICENSE](https://github.com/choderalab/asapdiscovery/blob/main/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

### CODE REVIEW

**Comment by JenkeScheen** [comment]
MVP of NS2B3 fitness view is done. Now I just need to go back and fix the broken subpocket view, because I had to make the 3DMol viewer chainID-dependent. I'll also extend the fitness view with some easy feature requests I get from P1/P2 feedback; will spawn issues for more advanced ones. Example view is here: https://a7e363c69a86a9f51702101656b2932994f870c5ddfeed52d1214bcaaacc.s3.amazonaws.com/fitness_ns2b3_demo/html_viz_fitness.html

Note that we're now baking in `base64` images of logo plots, which makes the HTML files a bit heavier (5-10MB) but still as portable.

**Comment by JenkeScheen** [comment]
- [ ] fix subpocket view by making it chainID-dependent
- [ ] check other fitness targets (Mpro + Mac1) and fix any issues
- [ ] check that all tests work
- [ ] incorporate some light-weight feature requests for the fitness view

**Comment by JenkeScheen** [comment]
ready for a first review pass @hmacdope, I'll add some small tweaks to phrasing in the legend after getting feedback from Karla. 

Note that there now are some gross workarounds in `html_viz.py` and `fitness.py` that deal with chain-dependent residue-index matching for the fitness view, I'm planning to remove this once I find the time to refactor this to do an automated alignment between the queried protein sequence and the queried DMS sequence.

File: asapdiscovery-dataviz/asapdiscovery/dataviz/html_viz.py [hunk 1]
```diff
@@ -570,3 +769,81 @@ def make_poses_html(self):
...
+                    font_name="Sans Serif",
+                    figsize=(3, 10),
+                    color_scheme="dmslogo_funcgroup",
+                    flip_below=False,
+                    show_spines=True,
+                )
+                plt.xticks([])
+                plt.yticks([])
+
+                # we could get base64 from buffer, but easier to write as tmp and read back as bas64.
```
Comment (hmacdope) [inline asapdiscovery-dataviz/asapdiscovery/dataviz/html_viz.py:838]: would prefer in memory if possible, but this module is full of tmpdir stuff anyway, so lets just roll with it.

**Review by hmacdope — CHANGES_REQUESTED** [review]
Looking good, some first comments. :)

File: asapdiscovery-data/asapdiscovery/data/fitness.py [hunk 2]
```diff
@@ -203,8 +209,14 @@ def get_fitness_scores_bloom_by_target(target: TargetTags) -> pd.DataFrame:
     # read the fitness data into a dataframe
     with open(fitness_data) as f:
         data = json.load(f)
-    data = data["data"]
-    fitness_scores_bloom = pd.DataFrame(data)
+    if "data" in data.keys():
+        data = data["data"]
+        fitness_scores_bloom = pd.DataFrame(data)
+    elif "ZIKV NS2B-NS3 (Closed)" in data.keys():
```
[the remark this answers is not in the document]
Comment (JenkeScheen) [reply asapdiscovery-data/asapdiscovery/data/fitness.py:216]: https://github.com/choderalab/asapdiscovery/pull/680/commits/9cb1dfa9d480c7a6c2eacee1d37a2271bfe6d128

File: asapdiscovery-dataviz/asapdiscovery/dataviz/html_viz.py [hunk 3]
```diff
@@ -6,21 +6,34 @@
 from pathlib import Path
 from typing import Dict, Optional, Union  # noqa: F401
 
+import pandas as pd
+
+pd.options.mode.chained_assignment = None
```
[the remark this answers is not in the document]
Comment (JenkeScheen) [reply asapdiscovery-dataviz/asapdiscovery/dataviz/html_viz.py]: yeah you're right.. https://github.com/choderalab/asapdiscovery/pull/680/commits/92e112a4fe7446ae11a804000e63337cdeea8aa9

File: asapdiscovery-dataviz/asapdiscovery/dataviz/html_viz.py (hunk 1 above)
Comment (JenkeScheen) [reply asapdiscovery-dataviz/asapdiscovery/dataviz/html_viz.py:835]: yeah it's not ideal, but I found some examples (e.g. [this one](https://stackoverflow.com/questions/66324439/matplotlib-image-to-base64-without-saving)) that would make this section less readable imo.

**Review by hmacdope — APPROVED** [review]
Looks good to me @JenkeScheen! Just fix CI and good to go.

**Pull request merged**